src/org/usfirst/frc4904/robot/commands/HopperAgitate.java

```diff
@@ -0,0 +1,31 @@
+package org.usfirst.frc4904.robot.commands;
+
+import org.usfirst.frc4904.robot.RobotMap;
+import org.usfirst.frc4904.robot.subsystems.Hopper;
+
+import edu.wpi.first.wpilibj.command.Command;
+
+public class HopperAgitate extends Command {
+	protected long lastSwitchTime;
+
+	public HopperAgitate() {
+		requires(RobotMap.Component.hopper);
+	}
+
+	@Override
+	protected void execute() {
+		if (System.currentTimeMillis() - lastSwitchTime > Hopper.AGITATE_DELAY) {
```

Comment:
There's one thing I would say — let's say the time for agitation has passed, but they have just turned. You might want to make it in the controls that when they shift the hopper on purpose, it resets the lastSwitchTime. Regardless, talk with @c0d3rman and/or Nathan about how they want their control scheme to work — it might be confusing during a match if the hopper is turned in the way they don't want it to be.